sdk/program/src/vote/state/mod.rs

```diff
@@ -247,6 +264,11 @@ pub struct VoteState {
 
     /// most recent timestamp submitted with a vote
     pub last_timestamp: BlockTimestamp,
+
+    /// Latencies of votes.  This VecDequeue parallels the votes member;
+    /// every entry in votes has a corresponding latency at the same
+    /// index here.
+    pub vote_latencies: VecDeque<u8>,
```

Comment:
This seems like it will break compatibility of snapshots between validators during the upgrade for folks that don't have/don't have this change, since all the the validators with this change will store a different struct in those snapshots.

Proper thing to do is probably to upgrade everyone with the new enums first, then switch people to storing the new version via a feature flag

Reply:
Also seems cleaner to fold this separate field into the existing `votes` struct, which can now be a `votes: VecDeque<LandedVote>`, where

```
struct LandedVote {
     latency: u8
     lockout: Lockout
}
```

This way we're guaranteed that `vote_latencies` and `vote` are the same length, have 1:1 mapping

Reply:
> This seems like it will break compatibility of snapshots between validators during the upgrade for folks that don't have/don't have this change, since all the the validators with this change will store a different struct in those snapshots.
> 
> Proper thing to do is probably to upgrade everyone with the new enums first, then switch people to storing the new version via a feature flag

Could make it an Option<VecDequeu<u8>>, which will naturally be None if not written.  Would that be OK?

Reply:
> Also seems cleaner to fold this separate field into the existing `votes` struct, which can now be a `votes: VecDeque<LandedVote>`, where
> 
> ```
> struct LandedVote {
>      latency: u8
>      lockout: Lockout
> }
> ```
> 
> This way we're guaranteed that `vote_latencies` and `vote` are the same length, have 1:1 mapping

That does more than add new data to the end of vote accounts, and I wasn't sure if it was OK to modify ABI frozen structs in the way you describe.  So it's OK to completely alter the encoding of a Vote account above and beyond appending to it?

Reply:
> I wasn't sure if it was OK to modify ABI frozen structs in the way you describe

Should be ok for since the enums describe which version we're using. Frozen ABI is just a reminder to make sure you think about backwards compatibility before checking in a change.

> So it's OK to completely alter the encoding of a Vote account above and beyond appending to it?

It would require more release hassle I think (i.e. your Option suggestion above won't work), but seems cleaner to maintain.